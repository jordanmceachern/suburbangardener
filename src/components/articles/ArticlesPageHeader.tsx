import React from "react";

interface ArticlesPageHeaderProps {
  className?: string;
}

export default function ArticlesPageHeader({
  className = "",
}: ArticlesPageHeaderProps) {
  return (
    <div className={`text-center mb-12 ${className}`}>
      <h1 className="text-4xl font-bold text-primary-900 dark:text-primary-100 mb-4">
        Gardening Articles
      </h1>
      <p className="text-lg text-neutral-600 dark:text-neutral-200 max-w-2xl mx-auto">
        Discover expert insights, seasonal tips, and in-depth guides for
        successful suburban gardening.
      </p>
    </div>
  );
}
