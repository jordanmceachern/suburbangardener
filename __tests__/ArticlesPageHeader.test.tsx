import { render } from "@testing-library/react";
import ArticlesPageHeader from "@/components/articles/ArticlesPageHeader";

describe("ArticlesPageHeader Component", () => {
  it("should match DOM snapshot", () => {
    const { container } = render(<ArticlesPageHeader />);
    expect(container.firstChild).toMatchSnapshot();
  });

  it("should maintain CSS classes for header layout", () => {
    const { container } = render(<ArticlesPageHeader />);

    // Check for header-related classes
    const hasHeaderClasses =
      container.innerHTML.includes("text-") ||
      container.innerHTML.includes("font-") ||
      container.innerHTML.includes("mb-") ||
      container.innerHTML.includes("text-center");

    expect(hasHeaderClasses).toBe(true);
  });
});
