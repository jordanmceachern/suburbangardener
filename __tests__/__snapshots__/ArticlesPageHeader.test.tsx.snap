// Jest Snapshot v1, https://jestjs.io/docs/snapshot-testing

exports[`ArticlesPageHeader Component should match DOM snapshot 1`] = `
<div
  class="text-center mb-12 "
>
  <h1
    class="text-4xl font-bold text-primary-900 dark:text-primary-100 mb-4"
  >
    Gardening Articles
  </h1>
  <p
    class="text-lg text-start text-neutral-600 dark:text-neutral-200 max-w-2xl mx-auto"
  >
    Discover expert insights, seasonal tips, and in-depth guides for successful suburban gardening.
  </p>
</div>
`;
